tools/lightning/filter.md

```diff
@@ -0,0 +1,118 @@
+---
+title: TiDB-Lightning Table Filter
+summary: Use black and white lists to filter out tables, ignoring them during import.
+category: tools
+---
+
+# TiDB-Lightning Table Filter
+
+TiDB-Lightning supports setting up black and white lists to ignore certain databases and tables. This can be used to skip cache tables, or manually partition the data source on a shared storage to allow multiple Lightning instances work together without interfering each other.
+
+The filtering rule is similar to MySQL `replication-rules-db`/`replication-rules-table`.
+
+## Filtering databases
+
+```toml
+[black-white-list]
+do-dbs = ["pattern1", "pattern2", "pattern3"]
+ignore-dbs = ["pattern4", "pattern5"]
+```
+
+* If the name of a database matched *any* pattern in the `do-dbs` array in the `[black-white-list]` section, the database is included.
+* Otherwise, if the name matched *any* pattern in the `ignore-dbs` array, the database is skipped.
+* If a database’s name matched *both* the `do-dbs` and `ignore-dbs` arrays, the database is included.
+
+The pattern can either be a simple name, or a regular expression in [Go dialect](https://golang.org/pkg/regexp/syntax/#hdr-Syntax) if it starts with a `~` character.
+
+## Filtering tables
+
+```toml
+[[black-white-list.do-tables]]
+db-name = "db-pattern-1"
+table-name = "table-pattern-1"
+
+[[black-white-list.do-tables]]
+db-name = "db-pattern-2"
+table-name = "table-pattern-2"
+
+[[black-white-list.do-tables]]
+db-name = "db-pattern-3"
+table-name = "table-pattern-3"
+
+[[black-white-list.ignore-tables]]
+db-name = "db-pattern-4"
+table-name = "table-pattern-4"
+
+[[black-white-list.ignore-tables]]
+db-name = "db-pattern-5"
+table-name = "table-pattern-5"
+```
+
+* If the qualified name of a table matched *any* pair of patterns in the `do-tables` array, the table is included.
+* Otherwise, if the qualified name matched *any* pair of patterns in the `ignore-tables` array, the table is skipped.
+* If a table’s qualified name matched *both* the `do-tables` and `ignore-tables` arrays, the table is included.
```

Comment:
add a rule `if the do-tables is not empty and no matched pair of patterns in it, the table is skipped`